src/core/qgsproject.h

```diff
@@ -575,6 +576,54 @@ class CORE_EXPORT QgsProject : public QObject, public QgsExpressionContextGenera
      */
     QMap<QString, QgsMapLayer *> mapLayers() const;
 
+    /**
+     * Unzip a project
+     * \param filename The zip file to unzip
+     * \returns true if unzip is well performed, false otherwise
+     * \since QGIS 3.0
+     */
+    bool unzip( const QString &filename );
+
+    /**
+     * Unzip a project with the current zip filename
+     * \returns true if unzip is well performed, false otherwise
+     * \since QGIS 3.0
+     */
+    bool unzip();
+
+    /**
+     * Zip the project
+     * \param filename The zip filename
+     * \returns true if zip is well performed, false otherwise
+     * \since QGIS 3.0
+     */
+    bool zip( const QString &filename );
+
+    /**
+     * Zip the project with the current zip filename
+     * \returns true if zip is well performed, false otherwise
+     * \since QGIS 3.0
+     */
+    bool zip();
+
+    /**
+     * Returns the current zip filename or an empty string if none.
+     * \since QGIS 3.0
+     */
+    QString zipFileName() const;
```

Comment:
Agreed with Nyall... I would even argue that client code should not have access to the temporary unpacked project file name - in theory right after load it could be removed again...?